<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [1]:
# Membuat dataframes dari dictionary
import pandas as pd
df = pd.DataFrame({'tipe_int': [50, 21], 'tipe_string': ['a', 'b']})

<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [3]:
# Membaca data dan memasukannya kedalam suatu variabel
diamonds = pd.read_csv("diamonds-1.csv")
print(diamonds)

       Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
0               1   0.23      Ideal     E     SI2   61.5   55.0    326  3.95   
1               2   0.21    Premium     E     SI1   59.8   61.0    326  3.89   
2               3   0.23       Good     E     VS1   56.9   65.0    327  4.05   
3               4   0.29    Premium     I     VS2   62.4   58.0    334  4.20   
4               5   0.31       Good     J     SI2   63.3   58.0    335  4.34   
...           ...    ...        ...   ...     ...    ...    ...    ...   ...   
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
0      3.98  2.43  

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab: Series adalah suatu objek satu dimensi yang menyimpan berbagai jenis tipe data.
Jika pada list dan dictionary, bisa berisi tipe data apa saja maka pada numpy array berupa tipe data yang homogen.
-Series adalah objek 1 dimensi
-DataFrame adalah suatu objek 2 dimensi, mirip seperti tabel

In [6]:
# code here
x = pd.Series([10,20,30,40])
print(x)

0    10
1    20
2    30
3    40
dtype: int64


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [23]:
# code here
print(diamonds.price.head(1))
print(diamonds.color.tail(1))

0    326
Name: price, dtype: int64
53939    D
Name: color, dtype: object


<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: suatu metode yang menghasilkan kesimpulan deskriptif statistik dari data.

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [44]:
# code here
print(diamonds.describe())
print(diamonds.price.mean())
print(diamonds.depth.std())
print(diamonds.carat.max())

         Unnamed: 0         carat         depth         table         price  \
count  53940.000000  53940.000000  53940.000000  53940.000000  53940.000000   
mean   26970.500000      0.797940     61.749405     57.457184   3932.799722   
std    15571.281097      0.474011      1.432621      2.234491   3989.439738   
min        1.000000      0.200000     43.000000     43.000000    326.000000   
25%    13485.750000      0.400000     61.000000     56.000000    950.000000   
50%    26970.500000      0.700000     61.800000     57.000000   2401.000000   
75%    40455.250000      1.040000     62.500000     59.000000   5324.250000   
max    53940.000000      5.010000     79.000000     95.000000  18823.000000   

                  x             y             z  
count  53940.000000  53940.000000  53940.000000  
mean       5.731157      5.734526      3.538734  
std        1.121761      1.142135      0.705699  
min        0.000000      0.000000      0.000000  
25%        4.710000      4.720000     

Jawab:
Rata - rata kolom price = 3932.799721913237
Standar deviasi kolom depth = 1.432621318833661
Nilai max kolom carat = 5.01